test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -188,86 +257,111 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(ovsCPUSet).To(Equal(onlineCPUSet))
 
+			if totalNumberofCgroupMembership == 2 {
+				Skip("CPU load balance can be checked only functionally on cgroupv2")
+			}
 			result, err = chkOvsCgroupLoadBalance(workerRTNode)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(result).To(Equal("0"))
 		})
 	})
 
 	Describe("Affinity", func() {
+		var ctx context.Context = context.TODO()
 		Context("ovn-kubenode Pods affinity ", func() {
 			testutils.CustomBeforeAll(func() {
+				profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+				Expect(err).ToNot(HaveOccurred())
 				initialProfile = profile.DeepCopy()
 			})
 			It("[test_id:64100] matches with ovs process affinity", func() {
-				testutils.KnownIssueJira("OCPBUGS-30806")
-				ovnPod, err := getOvnPod(context.TODO(), workerRTNode)
+				ovnPod, err := ovnCnfNodePod(ctx, workerRTNode)
 				Expect(err).ToNot(HaveOccurred(), "Unable to get ovnPod")
 
-				ovnContainersids, err := getOvnPodContainers(&ovnPod)
+				reservedCPU := string(*profile.Spec.CPU.Reserved)
+				reservedCPUSet, err := cpuset.Parse(reservedCPU)
+				Expect(err).ToNot(HaveOccurred())
+
+				ovnContainerids, err := ovnPodContainers(&ovnPod)
 				Expect(err).ToNot(HaveOccurred())
 
 				// Generally there are many containers inside a kubenode pods
 				// we don't need to check cpus used by all the containers
 				// we take first container
-				cpus := getCpusUsedByOvnContainer(context.TODO(), workerRTNode, ovnContainersids[0])
-				testlog.Infof("Cpus used by ovn Containers are %s", cpus)
-				pidList, err := getOVSServicesPid(context.TODO(), workerRTNode)
+				containerPid, err := nodes.ContainerPid(ctx, workerRTNode, ovnContainerids[0])
+				// we need to wait as process affinity can change
+				time.Sleep(30 * time.Second)
+				ctnCpuset := taskSet(ctx, containerPid, workerRTNode)
+				testlog.Infof("Cpus used by ovn Containers are %s", ctnCpuset.String())
+				pidList, err := ovsPids(ctx, ovsSystemdServices, workerRTNode)
 				Expect(err).ToNot(HaveOccurred())
-				cpumaskList, err := getCPUMaskForPids(context.TODO(), pidList, workerRTNode)
+				cpumaskList, err := getCPUMaskForPids(ctx, pidList, workerRTNode)
 				Expect(err).ToNot(HaveOccurred())
 				for _, cpumask := range cpumaskList {
-					Expect(cpus).To(Equal(cpumask), "affinity of ovn kube node pods(%s) do not match with ovservices(%s)", cpus, cpumask)
+					Expect(ctnCpuset).To(Equal(cpumask), "affinity of ovn kube node pods(%s) do not match with ovservices(%s)", ctnCpuset.String(), cpumask.String())
+					// The cpu affinity of pods and ovs services can match
+					// but we wanted to make sure that their affinity is not that of reserve cpus
```

Comment:
Why? Reserved will always be part of the mask.

Reply:
Right, on second thought i removed this test case , because i felt this is redundant, as i am already check if the containers of ovnkube-node pods have their affinity is set to all available online cpus, again adding another test case to check if affinity is not set reserved cpus only is redundant.

Reply:
+1